packages/cli/cloud/src/services/strapi-info-save.ts

```diff
@@ -5,12 +5,14 @@ import type { ProjectInfos } from './cli-api';
 export const LOCAL_SAVE_FILENAME = '.strapi-cloud.json';
 
 export type LocalSave = {
-  project?: ProjectInfos;
+  project?: Omit<ProjectInfos, 'id'>;
 };
 
-export async function save(data: LocalSave, { directoryPath }: { directoryPath?: string } = {}) {
-  const alreadyInFileData = await retrieve({ directoryPath });
-  const storedData = { ...alreadyInFileData, ...data };
+export async function save(
+  data: LocalSave,
+  { directoryPath, override }: { directoryPath?: string; override?: boolean } = {}
+) {
+  const storedData = override ? data : { ...(await retrieve({ directoryPath })), ...data };
```

Comment:
I believe it was more readable before, having the `await retrieve(...)` in another variable. 😅

Reply:
Why do we need the override parameter? Its already overwritten with how the object is created 🤔

Reply:
The save function in the original function merges the new data with the existing data. I don't want data from a previous config to coexist with new linked project data, so I've added an override parameter to re-use the function with the option of creating a brand new config object.

Reply:
Yes but it will overwrite any added data in future.
Also, the project data, as it is an object, will be replaced by the new one, no?

Reply:
1- Only if you pass the override parameter as true. Also for linking it works like that, you would like to have a brand new config when you link a new project to a folder with an existing config.
2- With the current implementation, the new project data will, but not the other keys of the object like node version and any other key you might have or want to add in the future.

Reply:
I see the point now, we are using shallow merges so the override is not needed in this case. Good catch, thanks.